### ISSUE TITLE
Layoutlmv2 - Document Classification predicting same class always

### ISSUE THREAD

**Issue by Jerome-Michael** [NielsRogge/Transformers-Tutorials#31]
Hi Niels Rogge,

Thanks for all the awesome tutorials.

I have been working on fine-tuning LayoutlmV2 model for document classification on my own data. I am facing an issue that the model is predicting same class for all examples, even for training examples. The model's training accuracy was above 90% after certain epochs , but it is always predicting the same class. I have changed the learning rate also, but no improvement. I have trained 400 examples for 3 classes, equally balanced. 

Another strange thing is , if I trained with less examples (5 to 15 examples for each class), it is predicting different classes, but if i train on more examples it is predicting same class.

Can you please help me regarding this ? Do I need to change any configuration before fine-tuning the model ?

Thanks in advance

Jerome

**Comment by NielsRogge** [comment]
> The model's training accuracy was above 90% after certain epochs , but it is always predicting the same class.

So you do have a very imbalanced dataset (i.e. a lot more samples of a given class compared to another class)?

**Comment by Jerome-Michael** [comment]
thanks for your reply.
I had a balanced dataset . I have solved the issue. The problem was the learning rate . I changed different learning rates and trained for longer epochs and now the model seems to predict different classes.

**Comment by Jerome-Michael** [comment]
the issue was solved and closed

**Comment by NielsRogge** [comment]
Always use 5e-5 to fine-tune Transformer-based models ;)

**Comment by Jerome-Michael** [comment]
yes sure ...thanks...

I have another question.

Can you please tell me how to load Layoutlxlm model (multilingual) ?

What is the tokenizer , feature extractor and processor for Layoutxlm ?

**Comment by NielsRogge** [comment]
You can leverage `LayoutLMv2FeatureExtractor` for LayoutXLM (as the image-related part is the same), however, the tokenizer for LayoutXLM is based on XLM-RoBERTa. So you would have to use the following tokenizer:

```
from transformers import AutoTokenizer

tokenizer = AutoTokenizer.from_pretrained("microsoft/layoutxlm-base")
```

So you would have to use both the feature extractor and tokenizer separately, there's no processor defined for it, for now.

**Comment by Jerome-Michael** [comment]
Thank you for the clear explanation @NielsRogge

**Comment by Laxmi530** [comment]
Hai @Jerome-Michael 
I am also facing the same issue can you please share what learning rate you used and how many classes you have in your dataset and how you balanced the dataset.

Thanking you in advance.

**Comment by Jerome-Michael** [comment]
Hi @Laxmi530 . 
I think I used 1e-5 . I trained for little more epochs around 50. I had 3 classes. totally balanced
